lib/json/common.rb

```diff
@@ -179,35 +301,148 @@ def parse!(source, opts = {})
+  #
+  # Bad Value:
+  #   # Raises TypeError (can't convert Symbol into Hash):
+  #   JSON.generate(obj, :foo)
+  #
+  # ====== Output Options
+  #
+  # The default formatting options generate the most compact
+  # \JSON data, all on one line and with no whitespace.
+  #
+  # You can use these formatting options to generate
+  # \JSON data in a more open format, using whitespace.
+  # See also JSON.pretty_generate.
+  #
+  # - Option +array_nl+ (\String) specifies a string (usually a newline)
+  #   to be inserted after each \JSON array; defaults to the empty \String, <tt>''</tt>.
+  # - Option +object_nl+ (\String) specifies a string (usually a newline)
+  #   to be inserted after each \JSON object; defaults to the empty \String, <tt>''</tt>.
+  # - Option +indent+ (\String) specifies the string (usually spaces) to be
+  #   used for indentation; defaults to the empty \String, <tt>''</tt>;
+  #   defaults to the empty \String, <tt>''</tt>;
+  #   has no effect unless options +array_nl+ or +object_nl+ specify newlines.
+  # - Option +space+ (\String) specifies a string (usually a space) to be
+  #   inserted after the colon in each \JSON object's pair;
+  #   defaults to the empty \String, <tt>''</tt>.
+  # - Option +space_before+ (\String) specifies a string (usually a space) to be
+  #   inserted before the colon in each \JSON object's pair;
+  #   defaults to the empty \String, <tt>''</tt>.
+  #
+  # In this example, +obj+ is used first to generate the shortest
+  # \JSON data (no whitespace), then again with all formatting options
+  # specified:
+  #
+  #   obj = {foo: [:bar, :baz], bat: {bam: 0, bad: 1}}
+  #   json = JSON.generate(obj)
+  #   puts 'Compact:', json
+  #   opts = {
+  #     array_nl: "\n",
+  #     object_nl: "\n",
+  #     indent+: '  ',
+  #     space_before: ' ',
+  #     space: ' '
+  #   }
+  #   puts 'Open:', JSON.generate(obj, opts)
+  #
+  # Output:
+  #   Compact:
+  #   {"foo":["bar","baz"],"bat":{"bam":0,"bad":1}}
+  #   Open:
+  #   {
+  #     "foo" : [
+  #       "bar",
+  #       "baz"
+  #   ],
+  #     "bat" : {
+  #       "bam" : 0,
+  #       "bad" : 1
+  #     }
+  #   }
+  #
+  # ---
+  #
+  # Raises an exception if any formatting option is not a \String.
+  #
+  # ====== Exceptions
+  #
+  # Raises an exception if +obj+ is not a valid Ruby object:
+  #   # Raises NameError (uninitialized constant Foo):
+  #   JSON.generate(Foo)
+  #   # Raises NameError (undefined local variable or method `foo' for main:Object):
+  #   JSON.generate(foo)
+  #
+  # Raises an exception if +obj+ contains circular references:
+  #   a = []; b = []; a.push(b); b.push(a)
+  #   # Raises JSON::NestingError (nesting of 100 is too deep):
+  #   JSON.generate(a)
+  #
+  # Raises an exception if +opts is not a
+  # {Hash-convertible object}[doc/implicit_conversion_rdoc.html#label-Hash-Convertible+Objects]
+  # (implementing +to_hash+):
+  #   # Raises TypeError (can't convert Symbol into Hash):
+  #   JSON.generate('x', :foo)
```

Comment:
Not useful, doc on argument is already very precise.

Reply:
Removed in next push.